mmf/modules/encoders.py

```diff
@@ -230,6 +241,50 @@ def forward(self, x):
         return out  # BxNx2048
 
 
+@registry.register_encoder("resnet")
+class ResNetEncoder(Encoder):
+    @dataclass
+    class Config(Encoder.Config):
+        name: str = "resnet"
+        backbone: str = None
+        pretrained: bool = True
+        pretrained_path: str = None
+
+    def __init__(self, config: Config, *args, **kwargs):
```

Comment:
If we can write simple tests to test the loading and output that would be great to make sure this doesn't break in future.